### ISSUE TITLE
Unable to take screenshot. WDA returned '{}' on iOS 11 real devices

### ISSUE THREAD

**Issue by rkelley11** [appium/appium#10481]
## The problem

Attempting to take a screenshot on any iOS 11 real device causes this issue.

## Environment

* Appium version (or git revision) that exhibits the issue:  1.7.2 & 1.7.1
* Last Appium version that did not exhibit the issue (if applicable): Unknown
* Desktop OS/version used to run Appium: macOS High Sierra 10.13.3
* Node.js version (unless using Appium.app|exe): 6.6.0
* Mobile platform/version under test: iOS 11.3 and iOS 11.1
* Real device or emulator/simulator: Real
* Appium CLI or Appium.app|exe: CLI

## Details

This is the first time trying to take screenshots, but this is working when I take screenshots with this code for devices with iOS 10 (uses Xcode 8.3.2).  I've updated my Xcode to the latest beta for 9.4 and selenium and appium packages with no luck.

## Link to Appium logs

https://gist.github.com/rkelley11/244e7456f126d622bdbd1d66ea148541

## Code To Reproduce Issue [ Good To Have ]
(just posting the lines that I think are relevant)
self.driver_package.mobile_driver = webdriver.Remote(appium_hub, desired_caps)
self.driver_package.mobile_driver.save_screenshot(file_and_path)

**Comment by mykola-mokhnach** [comment]
Does it work with appium@beta? Can you also attach the device log?

**Comment by rkelley11** [comment]
I uninstalled appium 1.7.2 and installed beta (v1.8.0-beta4) and got the same results.  I'm looking into how to get the device logs now and will post once I get them.

**Comment by rkelley11** [comment]
iPhone 7 Plus device log:  https://gist.github.com/rkelley11/5262fb7889a6e37bc05999864606444d

**Comment by rkelley11** [comment]
I searched the entire log and grabbed any line with XCTRunner in it:

https://gist.github.com/rkelley11/08f838a439736a562943c6401efecc07

**Comment by rkelley11** [comment]
I ran on an iPhone 6 with iOS 11.1.2 and had the same problem.  Here are the logs:

Device:  https://gist.github.com/rkelley11/52a38d69d6f4c84dcb5161dd8c9e0072

Appium:  https://gist.github.com/rkelley11/021bb53a27727321ff0dbcd599b7a873

**Comment by mykola-mokhnach** [comment]
I don't observe any explicit errors in the log. Might it be that the application under test or the device itself causes issues? What if you try to take a screenshot of another application or try it on an other device? Also, you could combine Appium server log with the device logs by setting `showXCodeLog` and `showIOSLog` capabilities to true.

**Comment by rkelley11** [comment]
The only other app I have to try on is an older version of our app (new version and the one used when first opening this issue was completely re-written from this version) and got the same results on three different devices:

iPhone 6 (iOS 11.1.2): https://gist.github.com/rkelley11/4f9a3baca41dc446313b6d67e21fe1d5
iPhone 7 (iOS 11):  https://gist.github.com/rkelley11/c69e38540d78195d3e3b7f5f5a944073
iPhone 7 Plus (iOS 11.3):  https://gist.github.com/rkelley11/ff80c6521a2be1b49738de94a21296ec

But tried on an iPhone 6 Plus (iOS 10.3.3) and it works fine.

**Comment by mykola-mokhnach** [comment]
Thanks for investigation. Try to also set `useNewWDA` to `true` to make sure the WDA binary is not cached. Also, try to minimise the app and take a screenshot while home screen is visible. I would also try the `element.getScreenshot` endpoint to check if screenshots can be at least taken from elements.

I'm almost sure it is some internal iOS issue, since `idevicescreenshot` tools also returns an error.

**Comment by rkelley11** [comment]
**Problem solved**

I tried useNewWDA = true and minimizing app and then taking screenshot and still had the issue.  Tried to do element.getScreenshot but it gave me an error that the method didn't exist.  I tried idevicescreenshot and got:

ERROR: Could not connect to lockdownd, error code -3

Found the same issue logged at https://github.com/libimobiledevice/libimobiledevice/issues/548

Followed what fordarnett did:

- uninstalled usbmuxd, ideviceinstaller, and libimobiledevice
- brew install --HEAD libimobiledevice

and it worked!

I confirmed it working on iPhone 7 Plus (iOS 11.3), iPhone 7 (iOS 11), iPhone 6 Plus (iOS 10.3.3) and iPhone 6 (iOS 11.1.2)

Thanks for all the help!
